### ISSUE TITLE
[Feature]: Support Python 3.13 in Compatibility Matrix

### ISSUE THREAD

**Issue by rishabh0510rishabh** [rishabh0510rishabh/EnvForage#142]
---
title: "[Feature]: Support Python 3.13 in Compatibility Matrix"
labels: ["type:feature", "compatibility-engine"]
---

### Description
Python 3.13 is now available, but our `PYTHON_MATRIX` and CLI probes currently stop at 3.12 (with only a minor probe for 3.13 in the CLI). We need to officially map framework support for 3.13 to allow users to generate modern environments.

### Files to Alter
- `backend/app/compatibility/matrix/python.py`
- `backend/seeds/profiles.yaml`

### Current Behavior
Frameworks are only mapped up to Python 3.12, and requests for 3.13 will fail compatibility checks.

### Expected Behavior
`PYTHON_MATRIX` should include Python 3.13 for compatible frameworks (e.g., latest PyTorch/TensorFlow versions), and seed profiles should be updated to allow it.

**Comment by ghost** [comment]
Hi @rishabh0510rishabh, I'd like to work on this under GSSoC 2026. I'll update backend/app/compatibility/matrix/python.py to include Python 3.13 framework mappings and update backend/seeds/profiles.yaml accordingly. The change is well-scoped and I can deliver a clean PR quickly. Please assign this to me.

**Comment by rishabh0510rishabh** [comment]
sure @soumyasekharshee265-ux

**Comment by ghost** [comment]
@rishabh0510rishabh  Thank you
